staker-rewards/lib/conditions.js

```diff
@@ -0,0 +1,180 @@
+import { callWithRetry } from "./contract-helper.js"
+
+export default class Conditions {
+  constructor(
+    context,
+    interval,
+    bondedECDSAKeepFactory,
+    keepBonding,
+    sortitionPoolAddress
+  ) {
+    this.context = context
+    this.interval = interval
+    this.bondedECDSAKeepFactory = bondedECDSAKeepFactory
+    this.keepBonding = keepBonding
+    this.sortitionPoolAddress = sortitionPoolAddress
+    this.operatorsDeauthorizedInInterval = []
+  }
+
+  static async initialize(context, interval) {
+    const { contracts } = context
+
+    const bondedECDSAKeepFactory = await contracts.BondedECDSAKeepFactory.deployed()
+
+    const sortitionPoolAddress = await callWithRetry(
+      bondedECDSAKeepFactory.methods.getSortitionPool(
+        contracts.sanctionedApplicationAddress
+      )
+    )
+
+    const conditions = new Conditions(
+      context,
+      interval,
+      bondedECDSAKeepFactory,
+      await contracts.KeepBonding.deployed(),
+      sortitionPoolAddress
+    )
+
+    return conditions
+  }
+
+  async checkDeauthorizations() {
+    // If Tenderly API is available refresh cached data.
+    if (this.context.tenderly) {
+      this.refreshDeauthorizationsCache()
+    }
+
+    // Fetch deauthorization transactions from cache.
+    // We expect only successful transactions to be returned, which means we don't
+    // need to double check authorizer vs operator as this was already handled
+    // by the contract on the function call.
+    const deauthorizations = this.context.cache.getFunctionCalls(
+      this.keepBonding.options.address,
+      "deauthorizeSortitionPoolContract"
+    )
+
+    console.debug(
+      `Found ${deauthorizations.length} sortition pool contract deauthorizations`
+    )
+
+    if (deauthorizations && deauthorizations.length > 0) {
+      for (let i = 0; i < deauthorizations.length; i++) {
+        const transaction = deauthorizations[i]
+
+        console.debug(`Checking transaction ${transaction.hash}`)
+
+        if (
+          transaction.block_number <= this.interval.startBlock ||
+          transaction.block_number >= this.interval.endBlock
+        ) {
+          console.debug(
+            `Skipping transaction made in block ${transaction.block_number}`
+          )
+          continue
+        }
+
+        const operator = transaction.decoded_input[0].value
+        const inputSortitionPool = transaction.decoded_input[1].value
```

Comment:
~~Why `inputSortitionPool` and not just `sortitionPool`?~~ OK, I do now. We have `sortitionPoolAddress` in the mix and we want to differentiate them. In this case, maybe `transactionOperator` and `transactionSortitionPool`?